jetty-core/jetty-core-server/src/test/java/org/eclipse/jetty/core/server/handler/EchoHandler.java

```diff
@@ -13,43 +13,49 @@
 
 package org.eclipse.jetty.core.server.handler;
 
+import java.util.concurrent.atomic.AtomicReference;
+
 import org.eclipse.jetty.core.server.Content;
 import org.eclipse.jetty.core.server.Handler;
 import org.eclipse.jetty.core.server.Request;
 import org.eclipse.jetty.core.server.Response;
 import org.eclipse.jetty.http.HttpHeader;
 import org.eclipse.jetty.util.Callback;
-import org.eclipse.jetty.util.IteratingCallback;
 import org.eclipse.jetty.util.StringUtil;
+import org.eclipse.jetty.util.thread.Invocable;
 
 /**
  * Dump request handler.
  * Dumps GET and POST requests.
  * Useful for testing and debugging.
  */
-public class EchoHandler extends Handler.Abstract
+public class EchoHandler extends Handler.Processor
 {
     @Override
-    public void handle(Request request) throws Exception
+    public void process(Request request, Response response, Callback callback)
     {
-        Response response = request.accept();
         response.setStatus(200);
         String contentType = request.getHeaders().get(HttpHeader.CONTENT_TYPE);
         if (StringUtil.isNotBlank(contentType))
             response.setContentType(contentType);
+        if (request.getHeaders().contains(HttpHeader.TRAILER))
+            response.getTrailers();
```

Comment:
Well there is a side effect of getting the trailers that prepares to response to accept trailers (ie forces chunking when it commits).
So you need to get them early before writing them.   I then get them again later rather than pass them around.   But that's a little wrong, so I'll pass the trailers into the Echo class. Standby....